lib/ecto/type.ex

```diff
@@ -89,15 +89,15 @@ defmodule Ecto.Type do
                       :binary | :decimal | :id | :binary_id |
                       :utc_datetime | :naive_datetime | :date | :time | :any |
                       :utc_datetime_usec | :naive_datetime_usec | :time_usec
-  @typep composite :: {:array, t} | {:map, t} | {:embed, Ecto.Embedded.t} | {:in, t}
+  @typep composite :: {:array, t} | {:map, t} | {:embed, Ecto.Embedded.t} | {:in, t} | {:tuple, [t]}
```

Comment:
We should probably handle this in the planner as we don't want to expose the tuple type to users. For example, we don't want to allow `field :foo, {:tuple, [:integer, :integer]}`.

Reply:
Is this different from `:in`? I assumed that `in` was just for `x.some_field in ^arr`, and that `tuple` could behave similarly.